flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/CustomizedConvertRule.java

```diff
@@ -0,0 +1,389 @@
+			return context.getRelBuilder().call(FlinkSqlOperatorTable.DATETIME_PLUS, childrenRexNode);
+		} else {
+			return context.getRelBuilder().call(FlinkSqlOperatorTable.PLUS, childrenRexNode);
+		}
+	}
+
+	private static RexNode convertReplace(CallExpression call, ConvertContext context) {
+		checkArgumentNumber(call, 2, 3);
+		List<Expression> children = call.getChildren();
+		List<RexNode> childrenRexNode = toRexNodes(context, children);
+		if (children.size() == 2) {
+			return context.getRelBuilder().call(
+				FlinkSqlOperatorTable.REPLACE,
+				childrenRexNode.get(0),
+				childrenRexNode.get(1),
+				context.getRelBuilder().call(FlinkSqlOperatorTable.CHAR_LENGTH, childrenRexNode.get(0)));
+		} else {
+			return context.getRelBuilder().call(FlinkSqlOperatorTable.REPLACE, childrenRexNode);
+		}
+	}
+
+	private static RexNode convertAs(CallExpression call, ConvertContext context) {
+		checkArgumentNumber(call, 2);
+		String name = extractValue((ValueLiteralExpression) call.getChildren().get(1), String.class);
+		RexNode child = context.toRexNode(call.getChildren().get(0));
+		return context.getRelBuilder().alias(child, name);
+	}
+
+	private static RexNode convertTrim(CallExpression call, ConvertContext context) {
+		checkArgumentNumber(call, 4);
+		List<Expression> children = call.getChildren();
+		ValueLiteralExpression removeLeadingExpr = (ValueLiteralExpression) children.get(0);
+		Boolean removeLeading = extractValue(removeLeadingExpr, Boolean.class);
+		ValueLiteralExpression removeTrailingExpr = (ValueLiteralExpression) children.get(1);
+		Boolean removeTrailing = extractValue(removeTrailingExpr, Boolean.class);
+		RexNode trimString = context.toRexNode(children.get(2));
+		RexNode str = context.toRexNode(children.get(3));
+		Enum trimMode;
+		if (removeLeading && removeTrailing) {
+			trimMode = SqlTrimFunction.Flag.BOTH;
+		} else if (removeLeading) {
+			trimMode = SqlTrimFunction.Flag.LEADING;
+		} else if (removeTrailing) {
+			trimMode = SqlTrimFunction.Flag.TRAILING;
+		} else {
+			throw new IllegalArgumentException("Unsupported trim mode.");
+		}
+		return context.getRelBuilder().call(
+			FlinkSqlOperatorTable.TRIM,
+			context.getRelBuilder().getRexBuilder().makeFlag(trimMode),
+			trimString,
+			str);
+	}
+
+	private static RexNode convertGet(CallExpression call, ConvertContext context) {
+		checkArgumentNumber(call, 2);
+		RexNode child = context.toRexNode(call.getChildren().get(0));
+		ValueLiteralExpression keyLiteral = (ValueLiteralExpression) call.getChildren().get(1);
+		Optional<Integer> indexOptional = ExpressionUtils.extractValue(keyLiteral, String.class).map(
+			child.getType().getFieldNames()::indexOf);
+		// Note: never replace the following code with :
+		// int index = indexOptional.orElseGet(() -> extractValue(keyLiteral, Integer.class));
+		// Because the logical in `orElseGet` always executed no matter whether indexOptional is present or not.
```

Comment:
I think this previous comment want to tell: The logical in `Optional.orElse` always executed no matter whether original optional is present or not. (So I change `isPresent` code to `orElseGet`)

`extractValue` will throw exception when extract fail, so we need avoid the evaluated here when the literal is string instead of integer.

I will remove this comment.